File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/comment/direct/Java/Python/s967693306_commented.py
import sys

def main():
    n = int(sys.stdin.readline())
    m = int(sys.stdin.readline())
    d = {}
    for _ in range(m):
        parts = sys.stdin.readline().split()
        s = int(parts[0])
        c = int(parts[1])
        if s in d:
            if d[s] != c:
                print(-1)
                return
        else:
            d[s] = c
    found = False
    for i in range(0, 1000):
        s_str = str(i)
        if len(s_str) < n:
            continue
        if len(s_str) != n:
            continue
        flag = True
        for j in range(n):
            pos = j + 1
            current_digit = int(s_str[j])
            if pos in d:
                if current_digit != d[pos]:
                    flag = False
                    break
            else:
                if j == 0:
                    if current_digit !=  dagen = int(sys.stdin.readline())
    m = int(sys.stdin.readline())
    d = {}
    for _ in range(m):
        parts = sys.stdin.readline().split()
        s = int(parts[0])
        c = int(parts[1])
        if s in d:
            if d[s] != c:
                print(-1)
                return
        else:
            d[s] = c
    found = False
    for i in range(0, 1000):
        s_str = str(i)
        if len(s_str) < n:
            continue
        if len(s_str) != n:
            continue
        flag = True
        for j in range(n):
            pos = j + 1
            current_digit = int(s_str[j])
            if pos in d:
                if current_digit != d[pos]:
                    flag = False
                    break
            else:
                if j == 0:
                    if current_digit != 1:
                        flag = False
                        break
                else:
                    if current_digit != 0:
                        flag = False
                        break
        if flag:
            print(s_str)
            return
    print(-1)

if __name__ == "__main__":
    main() # End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/comment/direct/Java/Go/s967693306_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	parts := strings.Fields(scanner.Text())
	n, _ := strconv.Atoi(parts[0])
	m, _ := strconv.Atoi(parts[1])

	constraints := make(map[int]int)
	conflict := false
	for i := 0; i < m; i++ {
		scanner.Scan()
		parts = strings.Fields(scanner.Text())
		s, _ := strconv.Atoi(parts[0])
		c, _ := strconv.Atoi(parts[1])

		existing, exists := constraints[s]
		if exists && existing != c {
			conflict = true
		}
		constraints[s] = c
	}

	if conflict {
		fmt.Println(-1)
		return
	}

	found := false
	var result string
	for i := 0; i < 1000; i++ {
		s := strconv.Itoa(i)
		if len(s) != n {
			continue
		}

		valid := true
		for j := 0; j < n; j++ {
			pos := j + 1
			currentDigit := int(s[j] - '0')

			requiredDigit, exists := constraints[pos]
			if exists {
				if currentDigit != requiredDigit {
					valid = false
					break
				}
			} else {
				if j == 0 && currentDigit == 1 {
					continue
				}
				if currentDigit != 0 {
					valid = false
					break
				}
			}
		}

		if valid {
			result = s
			found = true
			break
		}
	}

	if found {
		fmt.Println(result)
	} else {
		fmt.Println(-1)
	}
}

// End of Code
